### PR TITLE
[cmake] add PG_DLL

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -57,7 +57,7 @@ if( ${TARGET_OS} STREQUAL "Linux" )
 elseif(${TARGET_OS} STREQUAL "Windows" )
   set(OS "WINDOWS")
   set(OSNAME "Windows")
-  add_definitions(-DWIN32 -DHOST_WIN -DWINNT -DTARGET_WIN -DTARGET_WIN_X86)
+  add_definitions(-DWIN32 -DHOST_WIN -DWINNT -DTARGET_WIN -DTARGET_WIN_X86 -DPGDLL)
```
Comment (isuruf) [inline CMakeLists.txt:60]: This should only be defined for `_shared` targets only.

Comment (ghost) [reply CMakeLists.txt:60]: `flangmain` is compiled only once. Should we just add this to flangmain specifically?

Comment (ghost) [reply CMakeLists.txt:60]: See here: https://github.com/flang-compiler/flang/blob/a9ccdd766d5e606a1dc073fa3b106e52bab5dc64/runtime/flangmain/flangmain.c#L67

Comment (isuruf) [reply CMakeLists.txt:60]: flangmain is compiled as a static library, so it's not needed there.

**Pull request closed without merging**